Bugfix/103 ensure error code crawler is thread safe (#106)
exasol/error-code-crawler-maven-plugin: doc/design.md

An alternate solution special for deloboked code would be to replace the prefix of the generated directory with the original source directory. However then the line number would not fit, since Lombok expands the annotations. That could again be solved by searching in the original source for the error tag and by that determining the line number. However, we decided against this solution since for now the source position is not important enough.

Needs: impl, utest

### Maven Plugin should be Thread-Safe

`dsn~mvn-plugin-thread-safe~1`

> In order to support parallel execution of the maven plugin and remove WARNING messages during parallel execution, maven plugin should be marked as thread-safe

Reviewer: ```suggestion
In order to support parallel execution of the maven plugin and remove WARNING messages during parallel execution, maven plugin should be marked as thread-safe.
```

> 

Reviewer: Please add a rationale why it's ok to mark it as thread-safe. Link to the Maven documentation and list for each criteria why it's fulfilled.
E.g.: "... Mojo does store state in static fields ..."

An alternate solution special for deloboked code would be to replace the prefix of the generated directory with the original source directory. However then the line number would not fit, since Lombok expands the annotations. That could again be solved by searching in the original source for the error tag and by that determining the line number. However, we decided against this solution since for now the source position is not important enough.

Needs: impl, utest

### Maven Plugin should be Thread-Safe

`dsn~mvn-plugin-thread-safe~1`

In order to support parallel execution of the maven plugin and remove WARNING messages during parallel execution, maven plugin should be marked as thread-safe.
The following checklist was used to make sure that plugin is thread-safe (see https://cwiki.apache.org/confluence/display/MAVEN/Parallel+builds+in+Maven+3 for more details):

> * Checked that all static fields/variables in plugin/plugin code are not subject to threading problems.
> * Special attention was paid to find mutable static member variables of not thread-safe classes.
> * No mutable static member variables of not thread-safe classes was found.
> * Checked thread safety of any other third party libraries and made sure that they don't have mutable static member variables of not thread-safe classes.
> * Checked any plexus components.xml; if the components defined are singletons they need to be thread-safe (Nothing was found).
> * Checked for presence of known tainted libraries (Nothing was found.

Reviewer: ```suggestion
* Checked that no static fields/variables in plugin/plugin code are subject to threading problems.
* No mutable static member variables of not thread-safe classes were found.
* Checked thread safety of any other third party libraries and made sure that they don't have mutable static member variables of not thread-safe classes.
* The plugin does not use singleton plexus components in `components.xml`.
* The plugin does not use any known tainted libraries.
```
